src/components/InstallServiceworker.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component, Fragment } from 'react'
+import { string } from 'prop-types'
+import { Helmet } from 'react-helmet'
+
+import pick from '../utils/pick'
+import mapKeys from '../utils/mapKeys'
+import commonPropTypes from './commonPropTypes'
+
+class InstallServiceWorker extends Component {
+  static propTypes = {
+    src: string.isRequired,
+    dataIframeSrc: string,
+    dataNoServiceWorkerFallbackUrlMatch: string,
+    dataNoServiceWorkerFallbackShellUrl: string
+  }
+
+  static defaultProps = {
+    dataIframeSrc: undefined,
+    dataNoServiceWorkerFallbackUrlMatch: undefined,
+    dataNoServiceWorkerFallbackShellUrl: undefined
+  }
+
+  get properties() {
+    return mapKeys(pick(this.props, ...[
+      ...Object.keys(commonPropTypes),
```

Comment:
Seems that `amp-install-service-worker` supports only the `layout` from the common properties, so let's not pass all the common ones.